### ISSUE TITLE
Usage of MutationObserver causes animation deadlock with FontAwesome JS API

### ISSUE THREAD

**Issue by oxyc** [krokedil/klarna-checkout-for-woocommerce#537]
### Description of the issue

With the addition of https://github.com/krokedil/klarna-checkout-for-woocommerce/commit/6bc28f27e14147a0d45b249278eefccb7625d35d there's now multiple scrollTo animations queued if eg. FontAwesome `dom.watch()` is used on the checkout page.

### Expected behavior

Only one scroll should be triggered.

### Actual behavior

Multiple scrollTop animations are triggered entirely freezing the screen.

### Steps to reproduce the behavior

Add FontAwesome JS library with `dom.watch()` or I assume any other MutationObserver that changes the content.

### Klarna plugin version

2.10.2

### WordPress version

Latest

### WooCommerce version

Latest

### Screenshots

<img width="673" alt="Screenshot 2023-06-02 at 14 07 47" src="https://github.com/krokedil/klarna-checkout-for-woocommerce/assets/302736/30a3eaab-224c-4188-809d-03e3ba118dd0">

**Comment by oxyc** [comment]
Patch I used to fix it locally (havent really checked if the scroll actually works properly though)

```js.patch
diff --git a/assets/js/klarna-checkout-for-woocommerce.js b/assets/js/klarna-checkout-for-woocommerce.js
index 47a50b4..a561af9 100644
--- a/assets/js/klarna-checkout-for-woocommerce.js
+++ b/assets/js/klarna-checkout-for-woocommerce.js
@@ -59,7 +59,7 @@ jQuery( function( $ ) {
 						if (!$('html').hasClass(modalClassName)) {
 							// Wait for the Klarna modal to disappear before scrolling up to show error notices.
 							const noticeClassName = kco_params.pay_for_order ? 'div.woocommerce-notices-wrapper' : 'form.checkout';
-							$('html, body').animate({
+							$('html, body').not(':animated').animate({
 								scrollTop: ($(noticeClassName).offset().top - 100)
 							}, 1000);
 
```

**Comment by mntzrr** [comment]
Hello @oxyc!

Thanks for highlighting this issue. As you've pointed out, the action shouldn't be repeatedly triggered. However, I tried out your patch, and it didn't break the existing solution. Can you please create a PR so that we may attribute these changes to you?

**Comment by mntzrr** [comment]
I'll be closing this as not planned, since we do not have the proper means to reproduce this issue, and thus cannot verify the fix. However, feel free to create a PR so that we may attribute these changes to you!
